### PR TITLE
Build distroless proxy_init image with go version of istio-iptables

### PR DESCRIPTION
**Pull request by kramerul**
On step of plan in #12122

### CODE REVIEW

File: pilot/docker/Dockerfile.proxy_init [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+
+# The following section is used as base image if BASE_DISTRIBUTION=distroless
+FROM gcr.io/distroless/cc as distroless
+
+COPY --from=default /sbin/xtables-multi /sbin/iptables* /sbin/ip6tables* /sbin/ip /sbin/
+COPY --from=default /lib/xtables/ /lib/xtables
+COPY --from=default /lib/x86_64-linux-gnu/ /lib/x86_64-linux-gnu
+COPY --from=default /etc/iproute2 /etc/iproute2
+
+ADD istio-iptables /usr/local/bin/istio-iptables.sh
```
Comment (c0d1ngm0nk3y) [inline pilot/docker/Dockerfile.proxy_init]: So we name the binary `.sh`, right? Probably to reuse the `ENTRYPOINT`? I think having a go binary called `istio-tables.sh` is a bit confusing. Is there another possibility? Maybe using a link as `ENTRYPOINT` and letting this link point to the go binary or the script.

Comment (kramerul) [reply pilot/docker/Dockerfile.proxy_init]: I changed this

**Comment by jkbschmid** [comment]
@howardjohn Mind giving this a review? :)

**Pull request merged**